### PR TITLE
Added sample InfoWindowConcurrentActivity

### PR DESCRIPTION
**Pull request by zugaldia**
To illustrate the behavior of multiple InfoWindows visible at a time (https://github.com/mapbox/mapbox-gl-native/issues/3115).

### CODE REVIEW

**Comment by incanus** [comment]
This as well as the Info Window tracking to pans & zooms is :ok_hand: :100:

**Comment by zugaldia** [comment]
@incanus right. both features ended up being implemented at pretty much the same time.

**Comment by incanus** [comment]
Looking good! 

anigif-1448918357

One weird manifestation of this odd UI pattern is the final frame here: 

screen shot 2015-11-30 at 1 20 18 pm

Layering was done by which was above the other pre-rotation, but later rotations can throw this off.

**Comment by zugaldia** [comment]
cc: @tobrun

**Comment by tobrun** [comment]
@zugaldia 
Not really sure what best UX would be in this case.. hence this is an `odd ui pattern`.

In general: 
The parent layout is a FrameLayout so it will stack up its children in the way they were added to the ViewGroup. We could play around with this by using addView with an index or replace FrameLayout with a Relative so we can use bringToFront.

**Comment by zugaldia** [comment]
I have added the layering issue @incanus has found to the original ticket (https://github.com/mapbox/mapbox-gl-native/issues/3115#issuecomment-161065419) but this doesn't affect this PR, as it's just the test app and the basic functionality is there.

Also, the CI fail above is unrelated to Android, so if it all looks good otherwise, we're ready to merge the branch.

/cc: @bleege

**Comment by incanus** [comment]
> hence this is an `odd ui pattern`

File: android/MapboxGLAndroidSDKTestApp/src/main/res/layout/activity_infowindow_concurrent.xml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    <android.support.v7.widget.Toolbar
+        android:id="@+id/toolbar"
+        android:layout_width="match_parent"
+        android:layout_height="?attr/actionBarSize"
+        android:background="@color/primary"
+        android:theme="@style/ThemeOverlay.AppCompat.Dark.ActionBar" />
+
+    <com.mapbox.mapboxsdk.views.MapView
+        android:id="@+id/mapView"
```
Comment (bleege) [inline android/MapboxGLAndroidSDKTestApp/src/main/res/layout/activity_infowindow_concurrent.xml]: @zugaldia Let's change the id to something like `@+id/infoWindowConcurrentMapView`.  While there shouldn't be any runtime issues due to namespacing, I unfortunately got us into the habit of using `@+id/mapView` in all the TestApp examples which makes immediate sight recognition more difficult than it should be.

**Comment by zugaldia** [comment]
@bleege sounds good, I'll include that before merging.

**Pull request merged**